app/models/contribution.rb

```diff
@@ -0,0 +1,43 @@
+# Contributions made by a supporter to a project
+class Contribution < ApplicationRecord
+  attr_reader :amount_in_dollars
+  attr_writer :payment_processor
+  attr_accessor :charge
+
+  belongs_to :project
+  delegate :title, to: :project, prefix: true
+  delegate :statement_descriptor, to: :project
+
+  belongs_to :contributor, class_name: :RegisteredUser
+
+  attribute :amount_cents, :integer
+
+  def process
+    transaction do
+      save
+      return false unless persisted?
+      self.charge = payment_processor.charge(amount: amount)
```

Comment:
I want to start persisting this in the database, perhaps as JSON?

Reply:
All of this feels really dangerous to me. I don't like the false sense of safety that transactional stuff in distributed systems provides -- there's a rework of the data model necessary here. Will throw myself at it. Alternately we can just not persist contributions yet -- that's what I was going for with having Contribution be ActiveModel at first, so that we could punt on the shape of that

Reply:
We need to be storing the charge ID somewhere so if stripe throws a fit later we can trace it back. That doesn't mean it needs to be in the same model as the contribution, heck it could just be in logs for all we really care ATM, but losing access to the charge ID will be a giant pain in the ass later.

That doesn't mean we need to be doing it in a "database transaction" per-se, I was trying to get ahead of a thing you were concerned about earlier; which is why I put this in a real database transaction.

Reply:
One way we could slice this is to "take contributions on faith" initially, then fill in the stripe payment processor bits. Happy to see what slices you come up with.